test/codegen-generated-classes-test/src/test/java/software/amazon/awssdk/services/IdentityResolutionOverrideTest.java

```diff
@@ -0,0 +1,214 @@
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.assertj.core.api.Assertions.fail;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.function.Consumer;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import software.amazon.awssdk.auth.credentials.AwsBasicCredentials;
+import software.amazon.awssdk.auth.credentials.AwsCredentials;
+import software.amazon.awssdk.auth.credentials.StaticCredentialsProvider;
+import software.amazon.awssdk.auth.signer.AwsSignerExecutionAttribute;
+import software.amazon.awssdk.awscore.AwsRequest;
+import software.amazon.awssdk.awscore.AwsRequestOverrideConfiguration;
+import software.amazon.awssdk.core.RequestOverrideConfiguration;
+import software.amazon.awssdk.core.SdkRequest;
+import software.amazon.awssdk.core.SelectedAuthScheme;
+import software.amazon.awssdk.core.interceptor.Context;
+import software.amazon.awssdk.core.interceptor.ExecutionAttributes;
+import software.amazon.awssdk.core.interceptor.ExecutionInterceptor;
+import software.amazon.awssdk.core.interceptor.SdkInternalExecutionAttribute;
+import software.amazon.awssdk.identity.spi.Identity;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.restjsonendpointproviders.RestJsonEndpointProvidersClient;
+import software.amazon.awssdk.services.restjsonendpointproviders.RestJsonEndpointProvidersClientBuilder;
+import software.amazon.awssdk.utils.CompletableFutureUtils;
+
+class IdentityResolutionOverrideTest {
+
+    private static final AwsCredentials BASE_CREDENTIALS =
+        AwsBasicCredentials.create("akid", "skid");
+
+    private static final AwsCredentials OVERRIDE_CREDENTIALS =
+        AwsBasicCredentials.create("akidOverride", "skidOverride");
+
+    private CapturingInterceptor capturingInterceptor;
+
+    @BeforeEach
+    public void setup() {
+        this.capturingInterceptor = new CapturingInterceptor();
+    }
+
+    @Test
+    void when_credentialsProviderIsOverridden_atRequestCreateTime_itIsUsed() {
+        RestJsonEndpointProvidersClient syncClient = syncClientBuilder().build();
+
+        assertThatThrownBy(() -> syncClient.allTypes(r -> r.overrideConfiguration(
+            c -> c.credentialsProvider(StaticCredentialsProvider.create(OVERRIDE_CREDENTIALS)))))
+            .hasMessageContaining("stop");
+
+        assertSelectedAuthSchemeBeforeTransmissionContains(OVERRIDE_CREDENTIALS);
+    }
+
+    // Changing the credentials provider in modifyRequest does not work in SRA identity resolution
+    // Identity is resolved in beforeExecution (and happens before user applied interceptors) and cannot
+    // be affected by execution interceptors.
+    @Test
+    void when_credentialsProviderIsOverridden_inExecutionInterceptor_modifyRequest_itIsNotUsed() {
```

Comment:
In some test cases I see "itIsNotUsed" and "thayAreNotUsed"
You mean to say the overridden credentials in the Interceptors and if they are not used then what credentials get used 
like 
when_user_Provides_ProfileCredendial_And_InterceptorOverrideCredential_thenProfileCredential gets used. something like this would be helpful to clarify

Reply:
I can update the language if this heading doesn't make sense